### PR TITLE
[DNM][NEW SPECIES] Grod. Also, a ton of species improvements.

### LINKED ISSUE #5554: My grood foot notes
## My notes

Few notes
1. having gloves, going 4 stance then back to 2 causes the gloves to drop
2. science goggles cant be worn, I imagine for chemists that would be a major issue.
3. lack of suit storage means grod cant carry guns with them as easily, mainly issue for sec(e-guns dont fit in backpacks anymore)

### PR DESCRIPTION
**Pull request by Kapu1178**
### **GROD SPRITEWORK: NAEVII, PIRILL, TROONTJELOLO**

Cleaning up the body of this PR, as its outdated. See [here](https://forums.beestation13.com/t/grods-new-unique-xeno-race-now-in-development/15051/) instead

**TO-DO**
- [x] Refactor hand icon generation code
- [x] get_hand_offsets_for_dir()
- [X] Crownspider basic implementation
- [X] Stance Swap basic implementation
- [x] Crownspider completion
- [x] Stance Swap completion
- [x] Species damage modifiers
- [x] Species clothing restrictions
- [ ] Grod sprite accessories
- [ ] Grod Body Assimilation
- [x] Grod Language
- [X] SPRITES: Grod bodyparts
- [x] SPRITES: Grod crown customization
- [x] SPRITES: Grod hats
- [x] SPRITES: Grod shoes
- [x] SPRITES: Grod gloves
- [ ] SPRITES: Grod misc
- [ ] SPRITES: Grod backpacks
- [ ] SPRITES: Grod misc

## Why It's Good For The Game
It's a fucking new, unique species. 

## Changelog
:cl: Kapu1178, Naevii, Pirill, Troontjelolo
add: The Grod species
imageadd: Grod sprites by Naevii, Pirill, and Troontjelolo
refactor: inhand icons
refactor: moved IPC chassis handling into handle_mutant_bodyparts()
refactor: species reagent handling is moved entirely into species/proc/handle_chemicals()
code: support for shifting inhands around more than 2 hands
code: added the ability for species to have their own examine injury text. Applied to IPCs.
del: Cleaned up IPC code comments, removed copy pastes from synth code.
fix: Certain features now update properly, like moth wings, on character creation
fix: change_number_of_hands() wont nuke extra hands anymore.
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
commits before 572d621 are inherited from the sprite sheets branch.

**Comment by zeskorion** [comment]
can crowns enter other species? can they ventcrawl?

**Comment by Kapu1178** [comment]
> 
> 
> can crowns enter other species? can they ventcrawl?

1. Yes. They will be able to assimilate other species into a grod body.

2. Yes. This is literally just a bitflag so it can be changed at any time.

File: code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -466,20 +466,24 @@
...
-/mob/living/carbon/human/change_number_of_hands(amt)
+/mob/living/carbon/human/change_number_of_hands(amt, bloody = FALSE)
 	var/old_limbs = held_items.len
 	if(amt < old_limbs)
-		for(var/i in hand_bodyparts.len to amt step -1)
+		for(var/i in hand_bodyparts.len to amt+1 step -1)
 			var/obj/item/bodypart/BP = hand_bodyparts[i]
-			BP.dismember()
+			if(bloody)
+				BP.dismember() //Buggy
```
Comment (Archanial) [inline code/modules/mob/inventory.dm]: Ok

Comment (Kapu1178) [reply code/modules/mob/inventory.dm]: it can cause limbs to get stuck in limbo, making arms you can no longer use even if you get a new arm.

**Review by Archanial — CHANGES_REQUESTED** [review]
Scrap the spider idea entirely.

It's stupid to give out changeling ability to normal races.

**Comment by Kapu1178** [comment]
> 
> 
> Scrap the spider idea entirely.
> 
> It's stupid to give out changeling ability to normal races.

image


no. Its a completely neutered version of the changeling ability.

**Comment by Archanial** [comment]
I don't care, it's changeling ability nonetheless. Going out of your body and escaping should only be reserved to few.

**Comment by Kapu1178** [comment]
> 
> 
> I don't care, it's changeling ability nonetheless. Going out of your body and escaping should only be reserved to few.

sucks to suck i guess, because its literally not an escape tool.

edit: Wait i didnt even push the crownspider build to the repo. You dont even know what it fucking does yet.

**Comment by Archanial** [comment]
I can deduce that much from your PR body and gif included in it

**Comment by Kapu1178** [comment]
> 
> 
> I can deduce that much from your PR body and gif included in it

the pr body you didnt even read the first time and had to ping me in discord "can they wear hardsuits"?

**Comment by BrumalLake** [comment]
image
rp guidelines already supplied

**Comment by GregTheEgg42069** [comment]
G r o d

**Comment by Naevii** [comment]
> 
> 
> I can deduce that much from your PR body and gif included in it

We believe that regardless of their crownspider ability, this is going to be well balanced. And i will try to explain you how below.
-Crownspider cannot escape the body during any state that would prevent normal action (death, stun, etc) and cannot anter any body that already contain brain.
-Grods cannot be defibrilated and clonned. The only way for them to get revived is if someone remove their crownspider and plant it into another brainless person or toss it into hydroponics. (or if they leave the body before they die and then find another body quickly enough)
-Crownspiders are extremely frail, and dying in state of crownspider mob(if you willingly leave the body) makes you unrecoveable. (the only real way to bring crownspider mob to life is to use lazarus injector into it).

To summ up.  It makes it the ultimate escape ability yes, but only if your situation proves right otherwise you might end up permamently removed from the round. Keep in mind you also cannot use that ability if you are in state of stun or death etc. 
Additionaly, headspiders from changeling is such a strong ability because it blinds persons near him when he leaves the body, grods will have nothing like that. so they will lack the confusion element in comparision to headspiders.

If our actions won't prove enough to balance it out, we can consider removing their ventcrawl ability and ability to use language in this form. Everything is still subject to change, including health and speed values.

We believe that we really are doing good job balancing out Grod and his abilities, and that he won't end up being stronger than human.

**Comment by ghost** [comment]
>The Crownspider is very frail and should only be used as a last resort, as Grod do not want other species knowing about their true nature.

Why point this out if it will be ignored regardless?

**Comment by Archanial** [comment]
The issue isn't really how weak you will make their spider form, the issue is them having that form.

It's obvious you don't want it removed since you've made it like their gimmick and made sprites but the ability should go and they still will be really unique (4 hands, stances).

Also, since they can't be revived they don't have much interaction with medbay and having to ask miners for Lazarus isn't the greatest idea here.

**Comment by Kapu1178** [comment]
> 
> 
> The issue isn't really how weak you will make their spider form, the issue is them having that form.
> 
> It's obvious you don't want it removed since you've made it like their gimmick and made sprites but the ability should go and they still will be really unique (4 hands, stances).
> 
> Also, since they can't be revived they don't have much interaction with medbay and having to ask miners for Lazarus isn't the greatest idea here.

they can be "revived" by having a brainless body to hop into. Genetics can mass produce those.

They have a planned secondary revival method involving planting the crown in Botany to grow a new body.

They do not automatically use their Last Resort on death, and it cannot be use while unable to act. Their main revival method will be transferring the brain to a brainless body.

**Comment by Naevii** [comment]
> 
> 
> The issue isn't really how weak you will make their spider form, the issue is them having that form.
> 
> It's obvious you don't want it removed since you've made it like their gimmick and made sprites but the ability should go and they still will be really unique (4 hands, stances).
> 
> Also, since they can't be revived they don't have much interaction with medbay and having to ask miners for Lazarus isn't the greatest idea here.

You are missinterpreting them completely. They do take inspiration from headspider but it originated from their lore and concept in general. calling them "crownspider" and simplifying them to headspider is for convenience purposes only. 

I disagree with your stances regarding medbay interactions. they were given unique method of revival and this unique method of leaving their body for the sole purpose of making their medbay needs different than humans (no defib, no clonning). We are not that far into development yet but i had plans of involving some more unique medbay interactions. (keep in mind that me and Kapu are the only one working on Grod development and we have limited time resources). 

Also keep in mind that the only time they would require lazarus injectors is when they leave their body willingly. (which is their masive downside and balance to using their crown form to begin with).

Additionally. I given a lot of thought into balance, lore and their concept, and I would appreciate if you prefered to get yourself familiar with the way we plan to make things work rather than jumping to conclusions without reading thread properly.

**Comment by Naevii** [comment]
> 
> 
> > The Crownspider is very frail and should only be used as a last resort, as Grod do not want other species knowing about their true nature.
> 
> Why point this out if it will be ignored regardless?

We are pointng this out because the way we plan Grods to be, they will be roleplay heavy species and their lore will be a significant part of their gameplay. Rewarding players with unique playstyle. We plan them to be entirely different experience from any other playable race. 

And this even their naming convention will be way different than standard naming conventions we have. So they will not be advised for beginner players.

**Comment by Archanial** [comment]
No, grods should not be revived by using soulless bodies. Medbay shouldn't rely on cloning to work.
Roleplay has nothing to do with you getting headspider as race.

**Comment by Kapu1178** [comment]
> 
> 
> No, grods should not be revived by using soulless bodies. Medbay shouldn't rely on cloning to work.
> Roleplay has nothing to do with you getting headspider as race.

conveniently skipping over the fact theyre gonna be able to regrow bodies at Botany.

Grod are meant to be a species where if you die, coming back is not easy. They don't need to be as easy to revive as humans.

**Comment by Naevii** [comment]
> 
> 
> No, grods should not be revived by using soulless bodies. Medbay shouldn't rely on cloning to work.
> Roleplay has nothing to do with you getting headspider as race.

except you are not meant to use clonning at all with this race ? 
all you need is humanized monkey from genetics and then implant their brain in. or ever better to toss grod into botany so they let them grow in tray.
Roleplay and lore is core for this character design and character gameplay. 

They are nothing like "headspider playable race"
And if you put minimum efford to get yourself familiar with our project you would understand this. But you jump to conclusions without reading stuff properly.

**Comment by Archanial** [comment]
Conveniently skipping fact you can regrow bodies in botany as anyone

**Comment by Naevii** [comment]
> 
> 
> Conveniently skipping fact you can regrow bodies in botany as anyone

Which is unused. and is less beneficial than clonning. 
They are being striped of other means of easier and convenient revival methods on purpose.

**Comment by Naevii** [comment]
> 
> 
> Very important question
> 
> If grod shoes are planned can I wear more than one pair

Due to code limitations no. Only one pair of shoes.

**Comment by williamcll** [comment]
What are the damage taken multipliers for this species in comparison to a human?

**Comment by Kapu1178** [comment]
> 
> 
> What are the damage taken multipliers for this species in comparison to a human?

TBD

**Comment by francinum** [comment]
I don't have the attention bandwidth to pay to this at the moment.

I have approved the creation of this PR in compliance with codebase policy. I do not license the validity of the idea beyond that point.

**Comment by BrumalLake** [comment]
they don't want anything to do with it anymore 😳 


> 
> 
> > Very important question
> > If grod shoes are planned can I wear more than one pair
> 
> Due to code limitations no. Only one pair of shoes.

and these grod shits shouldn't be able to wear shoes i hope or you can have fun with making it look right

**Comment by Kapu1178** [comment]
> they don't want anything to do with it anymore 😳
> 
> > > Very important question
> > > If grod shoes are planned can I wear more than one pair
> > 
> > 
> > Due to code limitations no. Only one pair of shoes.
> 
> and these grod shits shouldn't be able to wear shoes i hope or you can have fun with making it look right

they will visually cover all their feet.

**Comment by Zenithrium** [comment]
this seems really neat! excited for a new unique race

**Comment by Kapu1178** [comment]
BcWtWOKVbG

**Comment by Troontjelolo** [comment]
HOOOONK

(Discount GAS)

**Comment by Kapu1178** [comment]
Ready for initial testmerge. More sprites are to come. ~~Cloning will be removed after a few TMs.~~
I removed cloning

**Comment by Troontjelolo** [comment]
LETS FUCKING GOOOO
heylois

**Comment by Naevii** [comment]
double clicking on grod player to orbit it as a ghost is very buggy. ghost glitches. i tried to double click again and it crashed my game, didnt allow me to reconnect after.

**Comment by dog132-ss13** [comment]
you cannot double dual-wield as you can't swap hands when wielding something in two hands

**Comment by yyzsong** [comment]
If grod lose their lower two arms, switching to four-arms mode will make the upper two not work until you switch to the lower ones and then back again.

**Comment by JackieOTheWisp** [comment]
So far pressing e to put a non-wearable hat into a bag gives you the "your species can't wear this" message 16 times in a row 
Also can you make it where we can at least wear monocles I want to be a very dapper caterpillar
also also hovering over a clothing space like I'm going to put on an item passively spams me with the aforementioned "you can't wear this" message

**Comment by Kapu1178** [comment]
> 
> 
> you cannot double dual-wield as you can't swap hands when wielding something in two hands

Intentional behavior.

**Comment by Kapu1178** [comment]
> 
> 
> So far pressing e to put a non-wearable hat into a bag gives you the "your species can't wear this" message 16 times in a row Also can you make it where we can at least wear monocles I want to be a very dapper caterpillar also also hovering over a clothing space like I'm going to put on an item passively spams me with the aforementioned "you can't wear this" message

Need to refactor how that message is displayed.

**Comment by Kapu1178** [comment]
> 
> 
> double clicking on grod player to orbit it as a ghost is very buggy. ghost glitches. i tried to double click again and it crashed my game, didnt allow me to reconnect after.

Incorrect BYOND version

**Comment by BrumalLake** [comment]
.9 and 1.1 speed mod technically aren't 10% faster and slower respectively

**Comment by PrefabQuasar** [comment]
when inputting a clown name for a Grod, it still follows the default naming rules which do not allow numbers to be entered in names, making it impossible for clown grod to follow the naming convention

**Comment by Zenithrium** [comment]
This is probably all already known, but prosthetic limbs, perscription glasses (likely other glasses too) and masks all have strange issues with sprites. Prosthetic legs just appear as if you were a bipedal race, glasses end up where your head would be if you were a human-like species, and masks look like they have their up / down states swapped

**Comment by Kapu1178** [comment]
> 
> 
> when inputting a clown name for a Grod, it still follows the default naming rules which do not allow numbers to be entered in names, making it impossible for clown grod to follow the naming convention

This is oddly a clown-specific issue. Investigating.

**Comment by Kapu1178** [comment]
> 
> 
> This is probably all already known, but prosthetic limbs, perscription glasses (likely other glasses too) and masks all have strange issues with sprites. Prosthetic legs just appear as if you were a bipedal race, glasses end up where your head would be if you were a human-like species, and masks look like they have their up / down states swapped

Simply due to the sprites not being ready yet.

**Comment by Crossedfall** [comment]
In terms of technical performance, no major impactful issues. In that sense the TM was a success. Still a couple of bugs here-and-there but nothing server breaking 👍. Should be good for a longer test once the sprites are finished

**Comment by JackieOTheWisp** [comment]
> > So far pressing e to put a non-wearable hat into a bag gives you the "your species can't wear this" message 16 times in a row Also can you make it where we can at least wear monocles I want to be a very dapper caterpillar also also hovering over a clothing space like I'm going to put on an item passively spams me with the aforementioned "you can't wear this" message
> 
> Need to refactor how that message is displayed.

1.) When using hotkeys to put an item I cannot wear into the backpack it tells me I can't wear it 16 times,

2.) hovering over the clothing inventory in the corner while on a hand holding a non-wearable item also randomly gives this message, even when not clicking

3.) More a complaint, but we cannot wear monocles and I wanted to be a very dapper caterpillar.

**Comment by Naevii** [comment]
> > > So far pressing e to put a non-wearable hat into a bag gives you the "your species can't wear this" message 16 times in a row Also can you make it where we can at least wear monocles I want to be a very dapper caterpillar also also hovering over a clothing space like I'm going to put on an item passively spams me with the aforementioned "you can't wear this" message
> > 
> > 
> > Need to refactor how that message is displayed.
> 
> 1.) When using hotkeys to put an item I cannot wear into the backpack it tells me I can't wear it 16 times,
> 
> 2.) hovering over the clothing inventory in the corner while on a hand holding a non-wearable item also randomly gives this message, even when not clicking
> 
> 3.) More a complaint, but we cannot wear monocles and I wanted to be a very dapper caterpillar.

Monocle was not sprited yet.  But will be.

**Comment by Zenithrium** [comment]
oh! I don't know if this is like logistically possible given BYOND, but would there be a way to view what the crownspider looks like in Grod character creation? There were a bunch of alternate styles and I didn't get to see most of them

**Comment by Naevii** [comment]
> oh! I don't know if this is like logistically possible given BYOND, but would there be a way to view what the crownspider looks like in Grod character creation? There were a bunch of alternate styles and I didn't get to see most of them

Crownspider (the simple mob that leave the body) have only one style.  Alternative styles are for the Crown on grod head under the hat. You can preview them in character creation. They are all showcased with their respective names in forum post https://forums.beestation13.com/t/grods-new-unique-xeno-race-now-in-development/15051

**Comment by Zenithrium** [comment]
> > oh! I don't know if this is like logistically possible given BYOND, but would there be a way to view what the crownspider looks like in Grod character creation? There were a bunch of alternate styles and I didn't get to see most of them
> 
> Crownspider (the simple mob that leave the body) have only one style. Alternative styles are for the Crown on grod head under the hat. You can preview them in character creation. They are all showcased with their respective names in forum post https://forums.beestation13.com/t/grods-new-unique-xeno-race-now-in-development/15051

Ah I misunderstood then. Thanks!

**Comment by BrumalLake** [comment]
aren't all clowns prohibited from having numbers in their name regardless of species
>1.) When using hotkeys to put an item I cannot wear into the backpack it tells me I can't wear it 16 times,

i believe that's an issue with `mob_can_equip()` that i've encountered before
image

**Comment by Kapu1178** [comment]
> 
> 
> aren't all clowns prohibited from having numbers in their name regardless of species
> 
> > 1.) When using hotkeys to put an item I cannot wear into the backpack it tells me I can't wear it 16 times,
> 
> i believe that's an issue with `mob_can_equip()` that i've encountered before image

Fixed. Was due to my dumbass not adding a check for "disablewarning" or something

**Comment by mcmeiler** [comment]
Immediately noticed this.

Examining a person doesn't show the rough extent of damage.

Also doesn't show if they're asleep or drunk.

**Comment by Kapu1178** [comment]
> 
> 
> Immediately noticed this.
> 
> Examining a person doesn't show the rough extent of damage.
> 
> Also doesn't show if they're asleep or drunk.

Fixed! Runtime error

**Comment by williamcll** [comment]
image
dreamseeker_jsXXsW4Y98
Grods seem not be able to wear any hats other than the top hat that they come with
Also mask animation seems to be flipped.

**Comment by Kapu1178** [comment]
> 
> 
> image dreamseeker_jsXXsW4Y98 Grods seem not be able to wear any hats other than the top hat that they come with Also mask animation seems to be flipped.

1. Intentional
2. Bug. Will fix.

**Comment by rkvothe14** [comment]
Is it normal not spawning with a portable crew monitor as a paramedic grod? This was late join, if it's important.

**Comment by Kapu1178** [comment]
> 
> 
> Is it normal not spawning with a portable crew monitor as a paramedic grod? This was late join, if it's important.

Odd. Investigating.

**Comment by pricklytomato** [comment]
Grod hats need to be available in clothing vendors and all the department clothing vendors, or the selection of hats grods can wear need to be extended

**Comment by Kapu1178** [comment]
> 
> 
> Grod hats need to be available in clothing vendors and all the department clothing vendors, or the selection of hats grods can wear need to be extended

They are. The Testmerge hasnt been updated with the commit, 7c97ef9

**Comment by williamcll** [comment]
currently, switching arm stances doesn't seem to change your movement speed, you are always slower than a 2 armed person.

**Comment by Kapu1178** [comment]
> 
> 
> currently, switching arm stances doesn't seem to change your movement speed, you are always slower than a 2 armed person.

Incredibly odd. Investigating.

**Comment by utilbird** [comment]
Pixar DMCA incoming
image

**Comment by WhereAmO** [comment]
Hey so plopping a grod brain into a tray seems to be broken. I, as a grod, spidered and hopped into a hydroponics tray, where I promply sat for the rest of the round and it did nothing. The botanist kept it watered and fed, and also made it into a Gaia pot later on. 

After a while, I examined the pot and it says it's "Empty", even though, well, there's an entire brain in it. Also, as a seperate issue, if someone takes a shovel to a grod brain in a pot it completely removes the brain in its entirety, maybe it shouldn't do that and just do lots of damage to it and make it an item..?

Also also, previously, as a grod, I have hopped into a tray and had a new body in less than a minute, so I am wondering if maybe the genetics (as in the whole thing botany can do and manipulate plants with) of the Grod Cocoons are random and that can end up in plants that grow incredibly fast and plants that never grow?
Grods_Broke
.

**Comment by GregTheEgg42069** [comment]
Grod cannot smoke 0/10

**Comment by ike709** [comment]
...How's this going?

**Comment by Naevii** [comment]
> ...How's this going?

From my perspective the basics sprites needed for the species were almost entirely completed.
If i recall correctly we only needed duffel bags? 

But Kapu halted grod changes untill kapu limbs are there and working. (not sure how is that going)

**Comment by Kapu1178** [comment]
showtime

**Comment by zeskorion** [comment]
grod? more like dead pr

**Comment by Kapu1178** [comment]
IM WORKIN ON IT

**Comment by DingleFlop** [comment]
Soon™️

**Comment by Tsar-Salat** [comment]
RIP Grods(Coming 2021)

**Comment by DerpiestBruh** [comment]
remember what they took away from u.

**Pull request closed without merging**